### PR TITLE
Fix bin test

### PR DESCRIPTION
**Pull request by dominictarr**
@regular this fixes the weird behaviour I was getting with the bin test: https://github.com/ssbc/scuttlebot/issues/565
In the process of trying to understand the test, I also simplified it a bit. Now it just iterates through the various combinations of config file, cli opts, connecting to ws or net, for different hosts.

Although, I notice it doesn't actually cover the problem I was having, because most of the tests do not run the cli client, just check that a network connection can be made to sbot.

@regular on that note, did you see: https://github.com/ssbc/ssb-integration-tests

### CODE REVIEW

**Comment by cryptix** [comment]
hmm... on travis the tests still fail with `Error: unexpected end of parent stream` on `./bin getAddress device ...`. It sounds like the sbot is going away to soon?

File: test/bin.js [hunk 1]
```diff
@@ -87,90 +90,63 @@ function connect(port, host, cb) {
...
-  try_often(10, function work(cb) {
-    connect(9001, '::1', cb)
-  }, function done(err) {
-    t.error(err, 'Successfully connect eventually')
-    end()
+  ;[9002, 9001].forEach(function (port) {
+    ;[true, false].forEach(function (asConfig) {
+      var opts = {
+        host: host,
+        port: 9001,
```
Comment (regular) [inline test/bin.js]: `port` is hardcoded here, you probably meant to use the iterated values.

**Comment by regular** [comment]
I can confirm that locally for me everything passes.

**Comment by cryptix** [comment]
I get the same behavior that travis also shows...

```
...
try more: 0
not ok 33 sbot getAdress succeeds eventually
  ---
    operator: error
    expected: |-
      undefined
    actual: |-
      { [Error: Command failed: /home/cryptix/scuttlebot/bin.js getAddress device -- --host=127.0.0.1 --port=9001 --path /tmp/sbot_binjstest_1540830749360 --caps.shs wCIhD8hG/YN4x77/DPGDY0ZzvGLuF+yXWZ9LzRhw5L4=
/home/cryptix/scuttlebot/node_modules/muxrpcli/index.js:68
      throw err
      ^

Error: unexpected end of parent stream
    at PacketStream.destroy (/home/cryptix/scuttlebot/node_modules/packet-stream/index.js:71:7)
    at PacketStream.write (/home/cryptix/scuttlebot/node_modules/packet-stream/index.js:133:41)
    at /home/cryptix/scuttlebot/node_modules/muxrpc/pull-weird.js:58:44
    at /home/cryptix/scuttlebot/node_modules/pull-stream/sinks/drain.js:20:24
    at /home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:17:11
    at /home/cryptix/scuttlebot/node_modules/packet-stream-codec/index.js:95:24
    at next (/home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:14:9)
    at /home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:16:41
    at Object.cb (/home/cryptix/scuttlebot/node_modules/packet-stream-codec/index.js:102:18)
    at drain (/home/cryptix/scuttlebot/node_modules/pull-reader/index.js:39:14)
] killed: false, code: 1, signal: null, cmd: '/home/cryptix/scuttlebot/bin.js getAddress device -- --host=127.0.0.1 --port=9001 --path /tmp/sbot_binjstest_1540830749360 --caps.shs wCIhD8hG/YN4x77/DPGDY0ZzvGLuF+yXWZ9LzRhw5L4=' }
    at: <anonymous> (/home/cryptix/scuttlebot/test/bin.js:182:7)
    stack: |-
      Error: Command failed: /home/cryptix/scuttlebot/bin.js getAddress device -- --host=127.0.0.1 --port=9001 --path /tmp/sbot_binjstest_1540830749360 --caps.shs wCIhD8hG/YN4x77/DPGDY0ZzvGLuF+yXWZ9LzRhw5L4=
      /home/cryptix/scuttlebot/node_modules/muxrpcli/index.js:68
            throw err
            ^
      
      Error: unexpected end of parent stream
          at PacketStream.destroy (/home/cryptix/scuttlebot/node_modules/packet-stream/index.js:71:7)
          at PacketStream.write (/home/cryptix/scuttlebot/node_modules/packet-stream/index.js:133:41)
          at /home/cryptix/scuttlebot/node_modules/muxrpc/pull-weird.js:58:44
          at /home/cryptix/scuttlebot/node_modules/pull-stream/sinks/drain.js:20:24
          at /home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:17:11
          at /home/cryptix/scuttlebot/node_modules/packet-stream-codec/index.js:95:24
          at next (/home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:14:9)
          at /home/cryptix/scuttlebot/node_modules/pull-goodbye/node_modules/pull-stream/throughs/filter.js:16:41
          at Object.cb (/home/cryptix/scuttlebot/node_modules/packet-stream-codec/index.js:102:18)
          at drain (/home/cryptix/scuttlebot/node_modules/pull-reader/index.js:39:14)
      
          at ChildProcess.exithandler (child_process.js:276:12)
          at emitTwo (events.js:126:13)
          at ChildProcess.emit (events.js:214:7)
          at maybeClose (internal/child_process.js:915:16)
          at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
  ...
ok 34 should be equal

1..34
# tests 34
# pass  33
# fail  1
```

**Comment by cryptix** [comment]
AFAICT this was merged without 0a71319 and even with it I can't get the tests to pass on my system.

It looks like multiserv just opens a socket on the first non-private v6 addr while ssb-client tries to connect to v4 or `localhost`:

```
# sbot client should work without options
scuttlebot 13.2.2 /tmp/sbot_binjstest_1545050706715 logging.level:notice
my key ID: A4FKSiZQd2m/gMAxL7rmuZP9H3IofJW9j/B5IGO22bg=.ed25519
WARNING-DEPRECATION: ssb-links not installed as a plugin. If you are using git-ssb, ssb-npm or patchfoo please consider installing it
Listening on 2a04:4540:7102:7600:beee:7bff:fe8c:6ffc:43293 (multiserver net plugin)
Listening on :43294 (multiserver ws plugin)
ssb-friends: stream legacy api used
try more: 10
retry run 10
work(err): { Error: Command failed: /home/cryptix/scuttlebot/bin.js getAddress device --path /tmp/sbot_binjstest_1545050706715 --config /tmp/sbot_binjstest_1545050706715/config --caps.shs mix6SmXakX2pqTMmvuisTtc50BoUhb2335K5+hO/Es0=
Error: Could not connect to the scuttlebot server localhost:43293
Use the "server" command to start it.

    at ChildProcess.exithandler (child_process.js:276:12)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at maybeClose (internal/child_process.js:915:16)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:209:5)
  killed: false,
  code: 1,
  signal: null,
  cmd: '/home/cryptix/scuttlebot/bin.js getAddress device --path /tmp/sbot_binjstest_1545050706715 --config /tmp/sbot_binjstest_1545050706715/config --caps.shs mix6SmXakX2pqTMmvuisTtc50BoUhb2335K5+hO/Es0=' }
try more: 9
retry run 9
work(err): { Error: Command failed: /home/cryptix/scuttlebot/bin.js getAddress device --path /tmp/sbot_binjstest_1545050706715 --config /tmp/sbot_binjstest_1545050706715/config --caps.shs mix6SmXakX2pqTMmvuisTtc50BoUhb2335K5+hO/Es0=
Error: Could not connect to the scuttlebot server localhost:43293


```

**Comment by regular** [comment]
I guess it passes for me, because locally I have https://github.com/ssbc/ssb-config/pull/23 merged.

**Comment by christianbundy** [comment]
Is this still valid, or has the recent test refactor made this obsolete?

**Comment by dominictarr** [comment]
the test is still there... we should probably move it out into an integration tests module though

**Pull request merged**